# Projeto Final

Nome: Kaylanne Mesquita

## Pergunta de Pesquisa 

**Quais características (hipertensão, idade, doença cardíaca, glicose, status de fumante, IMC) são mais comuns entre as pessoas que sofrem AVC?**


aquestao de genetica pode influenciar bastante em pessoas que sofrem AVC, pessoas fumantes ha muito tempo tendem a ter maior risco de sofrerem o infarto, por questoes pulmonares. A maioria dos casos de infarto sao pessoas com mais de 60 ano, porem, eventualmente acontecem em pessoas mais novas, por fumarem mais cedo, como o cigarro tradicional ou vape ( cigarro eletronico ), que atualmente e um dos mais causadores de avc e morte pelo mundo. Pessoas com o imc alto, eu acredito que tendem a tr mais chances de ter um avc, por conta de doenças cardiacas, volume de massa e etc...
Doenças cardiacas podem aumentar os riscos por conta do coraçao comprometido que pode enviar coagulos de samgue para o cerebro ou bombea-lo com menos força.
 

In [2]:
import pandas as pd 
import matplotlib.pyplot as plt 
import seaborn as sns 

In [3]:
df = pd.read_csv("healthcare-dataset-stroke-data.zip")
df.head()

,id,gender,age,hypertension,heart_disease,ever_married,work_type,Residence_type,avg_glucose_level,bmi,smoking_status,stroke
0,9046,Male,67.0,0,1,Yes,Private,Urban,228.69,36.6,formerly smoked,1
1,51676,Female,61.0,0,0,Yes,Self-employed,Rural,202.21,NaN,never smoked,1
2,31112,Male,80.0,0,1,Yes,Private,Rural,105.92,32.5,never smoked,1
3,60182,Female,49.0,0,0,Yes,Private,Urban,171.23,34.4,smokes,1
4,1665,Female,79.0,1,0,Yes,Self-employed,Rural,174.12,24.0,never smoked,1


In [4]:
df.columns

Index(['id', 'gender', 'age', 'hypertension', 'heart_disease', 'ever_married',
       'work_type', 'Residence_type', 'avg_glucose_level', 'bmi',
       'smoking_status', 'stroke'],
      dtype='str')

Index(['id', 'gênero', 'idade', 'hipertensão', 'doença_cardiaca', 'já_casado', 'tipo_de_trabalho', 'tipo_de_residencia', 'nível_médio_de_glicose', 'imc', 'status_de_fumante', 'acidente_vascular_cerebral'], dtype='str')

## Classificação das variáveis associadas a perguntas de pesquisa 


In [5]:
df["smoking_status"].unique()

<ArrowStringArray>
['formerly smoked', 'never smoked', 'smokes', 'Unknown']
Length: 4, dtype: str

a coluna smoking status e uma variavel categorica e tem 4 categorias 

In [8]:
df["hypertension"].unique()

array([0, 1])

a colulna de hypertension e uma variavel categorica e ela posssue dois valores que sao true ou false, que a pessoa pode ter ou nao hipertensao 

In [9]:
df["age"].unique()

array([6.70e+01, 6.10e+01, 8.00e+01, 4.90e+01, 7.90e+01, 8.10e+01,
       7.40e+01, 6.90e+01, 5.90e+01, 7.80e+01, 5.40e+01, 5.00e+01,
       6.40e+01, 7.50e+01, 6.00e+01, 5.70e+01, 7.10e+01, 5.20e+01,
       8.20e+01, 6.50e+01, 5.80e+01, 4.20e+01, 4.80e+01, 7.20e+01,
       6.30e+01, 7.60e+01, 3.90e+01, 7.70e+01, 7.30e+01, 5.60e+01,
       4.50e+01, 7.00e+01, 6.60e+01, 5.10e+01, 4.30e+01, 6.80e+01,
       4.70e+01, 5.30e+01, 3.80e+01, 5.50e+01, 1.32e+00, 4.60e+01,
       3.20e+01, 1.40e+01, 3.00e+00, 8.00e+00, 3.70e+01, 4.00e+01,
       3.50e+01, 2.00e+01, 4.40e+01, 2.50e+01, 2.70e+01, 2.30e+01,
       1.70e+01, 1.30e+01, 4.00e+00, 1.60e+01, 2.20e+01, 3.00e+01,
       2.90e+01, 1.10e+01, 2.10e+01, 1.80e+01, 3.30e+01, 2.40e+01,
       3.40e+01, 3.60e+01, 6.40e-01, 4.10e+01, 8.80e-01, 5.00e+00,
       2.60e+01, 3.10e+01, 7.00e+00, 1.20e+01, 6.20e+01, 2.00e+00,
       9.00e+00, 1.50e+01, 2.80e+01, 1.00e+01, 1.80e+00, 3.20e-01,
       1.08e+00, 1.90e+01, 6.00e+00, 1.16e+00, 1.00e+00, 1.40e

In [16]:
df["age"].min()


0.08

In [17]:
df["age"].max()

82.0

a coluna age e uma variavel numerico que varia entre 0.08 e 82.0 diante desses valores podemos assumir que a unidade de medida e ano e quem tem menos de 1 sao meses

In [10]:
df["heart_disease"].unique()

array([1, 0])

a coluna heart disease e uma variavel categorica true ou false que a pessoa pode ter ou nao 

In [ ]:
df["bmi"].unique()

array([36.6,  nan, 32.5, 34.4, 24. , 29. , 27.4, 22.8, 24.2, 29.7, 36.8,
       27.3, 28.2, 30.9, 37.5, 25.8, 37.8, 22.4, 48.9, 26.6, 27.2, 23.5,
       28.3, 44.2, 25.4, 22.2, 30.5, 26.5, 33.7, 23.1, 32. , 29.9, 23.9,
       28.5, 26.4, 20.2, 33.6, 38.6, 39.2, 27.7, 31.4, 36.5, 33.2, 32.8,
       40.4, 25.3, 30.2, 47.5, 20.3, 30. , 28.9, 28.1, 31.1, 21.7, 27. ,
       24.1, 45.9, 44.1, 22.9, 29.1, 32.3, 41.1, 25.6, 29.8, 26.3, 26.2,
       29.4, 24.4, 28. , 28.8, 34.6, 19.4, 30.3, 41.5, 22.6, 56.6, 27.1,
       31.3, 31. , 31.7, 35.8, 28.4, 20.1, 26.7, 38.7, 34.9, 25. , 23.8,
       21.8, 27.5, 24.6, 32.9, 26.1, 31.9, 34.1, 36.9, 37.3, 45.7, 34.2,
       23.6, 22.3, 37.1, 45. , 25.5, 30.8, 37.4, 34.5, 27.9, 29.5, 46. ,
       42.5, 35.5, 26.9, 45.5, 31.5, 33. , 23.4, 30.7, 20.5, 21.5, 40. ,
       28.6, 42.2, 29.6, 35.4, 16.9, 26.8, 39.3, 32.6, 35.9, 21.2, 42.4,
       40.5, 36.7, 29.3, 19.6, 18. , 17.6, 19.1, 50.1, 17.7, 54.6, 35. ,
       22. , 39.4, 19.7, 22.5, 25.2, 41.8, 60.9, 23

In [15]:
df["bmi"].min()


10.3

In [14]:
df["bmi"].max()

97.6

a coluna bmi e uma variavel numerica que fica entre 10.3 e 97.6

## Limpeza de dados 

In [21]:
dados_faltantes_soma = df.isnull().sum()
dados_faltantes_porc = df.isnull().mean() * 100

df_dados_faltantes = pd.DataFrame({"Dados faltantes (soma)": dados_faltantes_soma,
                                  "Dados faltantes (porcentagem)": dados_faltantes_porc }) 
df_dados_faltantes

,Dados faltantes (soma),Dados faltantes (porcentagem)
id,0,0.000000
gender,0,0.000000
age,0,0.000000
hypertension,0,0.000000
heart_disease,0,0.000000
ever_married,0,0.000000
work_type,0,0.000000
Residence_type,0,0.000000
avg_glucose_level,0,0.000000
bmi,201,3.933464


In [24]:
df = df.dropna(subset=["bmi"])

eu analisei que na coluna "BMI" havia 3% de dados nulos por isso os apaguei.